README.md

```diff
@@ -77,6 +77,111 @@ You can pass any of the usual arguments to render in this call: `layout`, `statu
 
 *Note: you need to have [Webpack process your code](https://github.com/rails/webpacker#binstubs) before it is available to the browser, either by manually running `./bin/webpack` or having the `./bin/webpack-watcher` process running.*
 
+## Hot Module Replacement
+
+[HMR](https://webpack.js.org/guides/hmr-react/) allows to load changes to react components without reloading.
+
+1. install `react-hot-loader` and
+
+```
+./bin/yarn add react-hot-loader@3.0.0-beta.6 --dev
+./bin/yarn add webpacker_react --dev
+```
+
+2. perform the following changes to your webpack config
+
+```js
+{
+  output: {
+    publicPath: 'http://localhost:8080'
+  },
+  devServer: {
+    contentBase: path.resolve('..', 'public', 'packs'),
+    publicPath: webpackDevServerAddr
+  }
+}
+```
+
+add `react-hot-loader/babel` to your `babel-loader` rules
+
+```js
+
+{
+module: {
+  rules: [
+    {
+      test: /\.jsx?(.erb)?$/,
+      exclude: /node_modules/,
+      loader: 'babel-loader',
+      options: {
+        presets: [
+          'react',
+          [ 'latest', { 'es2015': { 'modules': false } } ]
+        ],
+        plugins: ["react-hot-loader/babel"]
+      }
+    }
+}
+
+```
+
+prepend `react-hot-loader/patch` to your entries
+
+```js
+{
+  entry:
+    { application: [ 'react-hot-loader/patch', '../app/javascript/packs/application.js' ],
+    ...
+}
+```
+
+we provied a convience function to add the neccesary changes to your config if it's not
+significanlty differnet thatn the standard webpacker config:
+
+```js
+//development.js
+...
+
+var webpackConfigForHMR = require('webpacker-react/webpack-config-for-hmr')
+
+var config = require('./shared.js')
+
+config = configureHotModuleReplacement(config);
+
+module.exports = merge(config, ...)
+
+```
+
+
+3. configure and start dev server:
+
+make sure the following line is in your development.rb
+
+```ruby
+config.x.webpacker[:dev_server_host] = 'http://localhost:8080/'
+```
+
+start dev server in hot replacement mode
+
+```sh
+./bin/webpack-dev-server --hot
+```
+
+4. opt in to HMR from your pack files
+
+```es6
+
+import SomeRootReactComponent from 'components/some-root-react-component';
+import WebpackerReact from 'webpacker-react/hmr';
+
+WebpackerReact.register(SomeRootReactComponent)
+if (module.hot) module.hot.accept('components/some-root-react-component', () => WebpackerReact.renderOnHMR(SomeRootReactComponent) )
```

Comment:
Dont use `;` (Webpacker styling convention), and add some line breaks, this is more readable in the README